services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -3156,11 +3161,26 @@ private void deleteOneStoreVersion(String clusterName, String storeName, int ver
           cleanUpViewResources(new Properties(), store, deletedVersion.get().getNumber());
         }
         if (store.isDaVinciPushStatusStoreEnabled()) {
-          getPushStatusStoreWriter().deletePushStatus(
-              storeName,
-              deletedVersion.get().getNumber(),
-              Optional.empty(),
-              deletedVersion.get().getPartitionCount());
+          ExecutorService executor = Executors.newSingleThreadExecutor();
+          Future<?> future = executor.submit(
+              () -> getPushStatusStoreWriter().deletePushStatus(
+                  storeName,
+                  deletedVersion.get().getNumber(),
+                  Optional.empty(),
+                  deletedVersion.get().getPartitionCount()));
+          try {
+            future.get(PUSH_STATUS_STORE_WRITER_TIMEOUT_MS, TimeUnit.MILLISECONDS);
+            LOGGER.info("Deleted da-vinci push status of {}_v{}", storeName, versionNumber);
+          } catch (InterruptedException e) {
+            LOGGER.error("Got interrupted when deleting da-vinci push status of {}_v{}", storeName, versionNumber);
```

Comment:
Swallowing interrupted exceptions is typically considered a bad practice... I think in Venice we mostly use the approach 4.3 mentioned in [this blog post](https://www.baeldung.com/java-interrupted-exception#3-custom-exception-handling), though feel free to pick any approach you prefer...

Reply:
Thanks for catching this! I adopted approach 4.2 in the blog post, because I wanna RT deletion (next step of the function) to be executed. If throwing an exception, potential RT deletion won't happen after this specific version deletion.